packages/core/test/integration/data-types/data-types.test.ts

```diff
@@ -1705,40 +1744,48 @@ describe('DataTypes', () => {
             // TODO: expected for mariadb 10.4 : https://jira.mariadb.org/browse/MDEV-15558
             expect(table.jsonStr.type).to.equal('LONGTEXT');
             break;
+          case 'oracle':
+            expect(table.jsonStr.type).to.equal('BLOB');
+            break;
           default:
             expect(table.jsonStr.type).to.equal(jsonTypeName);
         }
       });
 
-      it('properly serializes default values', async () => {
-        const createdUser = await vars.User.create();
-        await createdUser.reload();
-        expect(createdUser.get()).to.deep.eq({
-          jsonStr: 'abc',
-          jsonBoolean: true,
-          jsonNumber: 1,
-          jsonNull: null,
-          jsonArray: ['a', 'b'],
-          jsonObject: { key: 'abc' },
-          id: 1,
+      // Oracle Database < 21 doesn't consider scalars as JSON column
+      // thus, fails the CHECK constraint test.
+      if (dialect.name !== 'oracle') {
```

Comment:
Can we make this dependent on the Oracle version? And is there a similar test that we could do?